Integration test timing issue: first test in TestTracingMiddlewareSpanIngestion fails to ingest spans

## Issue
The `test_authorization_header_stripped` test in `tests/integration/test_audit_tracing_middleware.py` consistently fails due to a timing/initialization issue where the first test to run in the `TestTracingMiddlewareSpanIngestion` class cannot retrieve spans from storage, even though the span is created (trace_id is present in response headers).

## Failing Test
- `test_authorization_header_stripped` (TestTracingMiddlewareSpanIngestion)

## Error Details
```
AssertionError: Span not ingested
```

At line 509: `assert span, "Span not ingested"`

## Analysis

### Test Pattern
The test fails consistently as the **first test alphabetically** in `TestTracingMiddlewareSpanIngestion`. All subsequent tests that use the same endpoint (`/auth/v1/root/`) and wait for spans pass successfully:

**Test Execution Order:**
1. ❌ `test_authorization_header_stripped` - **FAILS** (first alphabetically)
2. ✅ `test_large_response_body_truncated` - PASSES (uses `/auth/v1/root/`)
3. ✅ `test_ingestion_is_async_non_blocking` - PASSES
4. ✅ `test_request_body_captured_for_supported_types` - PASSES (uses `/auth/v1/root/`)
5. ✅ `test_request_body_captured_for_form_data` - PASSES
6. ✅ `test_query_params_captured` - PASSES
7. ✅ `test_response_headers_captured` - PASSES (uses `/auth/v1/root/`)
8. ✅ `test_duration_ms_is_accurate` - PASSES
9. ✅ `test_span_is_recorded_on_request` - PASSES
10. ✅ `test_trace_id_echoed_in_response` - PASSES

### Key Observations

1. **Trace ID is generated**: The response includes `X-Request-ID` header, confirming the tracing middleware is processing the request
2. **Request succeeds**: HTTP 200 response, no errors in request handling
3. **Async ingestion issue**: The span is not retrievable from storage within the timeout period
4. **Subsequent tests pass**: Tests 2-10 that make identical requests successfully retrieve their spans
5. **Class-specific**: This only affects tests in `TestTracingMiddlewareSpanIngestion`, not `TestTracingMiddlewareBasic`

### Potential Root Causes

1. **Audit client initialization**: The first test may run before the audit client singleton is fully initialized
2. **Async executor state**: The `tracing._ingestion_executor` may not be ready to process tasks immediately after `setUpClass`
3. **Storage initialization**: Traces storage may not be fully ready despite `TracesResource.init_storage()` being called
4. **Race condition**: The first span ingestion may race with some initialization logic that completes by the time the second test runs

### Related Code

**Test Setup:**
```python
@classmethod
def setUpClass(cls):
    # ... initialization code ...
    cls._check_dependencies()  # Makes a test request, verifies span ingestion
    # Dependency check PASSES, but first test still FAILS

def setUp(self):
    # Called before each test
    TracesResource.init_storage()
    self.manager.reset_test_data()
    TracesResource.init_storage()
    tracing._audit_client = None  # Reset singleton
    traces_storage.delete_matching({})  # Clear storage
```

**Test Code:**
```python
def test_authorization_header_stripped(self):
    response = self.auth_client.post(
        "/auth/v1/root/",
        json={"client_id": env.CLIENT_ID, "client_secret": env.CLIENT_SECRET},
        headers=self.auth_headers,
    )
    trace_id = response.headers.get("X-Request-ID")
    span = self._wait_for_span(trace_id)  # Times out, returns None
    assert span, "Span not ingested"  # FAILS HERE
```

## Attempted Fixes (None Worked)

1. ✅ **Increased timeout** - Changed `_wait_for_span(trace_id)` to `_wait_for_span(trace_id, timeout=2.0)` - Still failed
2. ✅ **Warm-up request** - Added `self.auth_client.get("/test/health")` before test - Still failed
3. ✅ **Delay in setUp** - Added `time.sleep(0.1)` in setUp - Still failed
4. ✅ **Different endpoints** - Tried `/test/health`, `/auth/v1/root/` - Same issue
5. ✅ **Not resetting audit client** - Removed `tracing._audit_client = None` - Still failed

## Impact

- **Test reliability**: One test consistently fails, affecting CI/CD reliability
- **False positives**: The test failure is not due to the actual feature (Authorization header stripping) but a timing issue
- **Confusing**: The failure masks whether the actual feature under test works

## Workarounds

1. **Skip the test**: Add `@unittest.skip` decorator (loses test coverage)
2. **Rename test**: Change to `test_zzz_authorization_header_stripped` to run last (moves problem, doesn't fix it)
3. **Accept failure**: Consider this a known issue (not ideal)

## Suggested Fixes

1. **Investigate audit client initialization**: Ensure the audit client is fully initialized before the first test runs
2. **Add initialization delay in setUpClass**: Add a small delay after dependency check to ensure async executor is ready
3. **Improve dependency check**: Make the dependency check verify that the *next* request will also succeed
4. **Debug async executor**: Add logging to understand why the first span ingestion fails
5. **Refactor test setup**: Consider not resetting the audit client singleton between tests if it's causing initialization issues

## Context

- Related to issue #459 (async span ingestion)
- Issue #493 (non-existent endpoints) has been fixed
- This is a separate timing issue that affects the first test to run
- 30 out of 31 integration tests pass (96.8% pass rate)

## Test Results

```
Ran 31 tests in 2.113s
FAILED (failures=1)
```

Only `test_authorization_header_stripped` fails.

--- a/tests/integration/test_audit_tracing_middleware.py
+++ b/tests/integration/test_audit_tracing_middleware.py
@@ -240,6 +240,14 @@ class TestTracingMiddlewareSpanIngestion(DependencyCheckedTestCase):
         # executor ensures clean state for this dependency check.
         from campus.audit.middleware import tracing
         original_executor = tracing._ingestion_executor
+
+        # CRITICAL: Wait for the original executor to finish all pending tasks
+        # before replacing it. This prevents issue #496 where pending tasks from
+        # the previous test class are still processing when we create the fresh
+        # executor, causing those tasks to fail when the original executor is
+        # restored and shut down.
+        original_executor.shutdown(wait=True)
+
         tracing._ingestion_executor = concurrent.futures.ThreadPoolExecutor(
             max_workers=2, thread_name_prefix="dependency_check_ingest"
         )
@@ -293,8 +301,17 @@ class TestTracingMiddlewareSpanIngestion(DependencyCheckedTestCase):
 
         finally:
             # Clean up the temporary executor
+            # Use wait=True to ensure all pending tasks complete before shutdown
+            # This prevents issue #496 where spans fail to ingest with
+            # "cannot schedule new futures after shutdown" error
             tracing._ingestion_executor.shutdown(wait=True)
-            tracing._ingestion_executor = original_executor
+
+            # CRITICAL: Recreate the original executor instead of restoring the
+            # shut-down executor. We already shut down the original executor
+            # before creating the fresh one, so we need to recreate it here.
+            tracing._ingestion_executor = concurrent.futures.ThreadPoolExecutor(
+                max_workers=2, thread_name_prefix="audit_ingest"
+            )
 
     @classmethod
     def tearDownClass(cls):

